docs/content/sips/xxx-spin-plugins.md

```diff
@@ -0,0 +1,196 @@
+title = "SIP xxx - Plugin System for Spin"
+template = "main"
+date = "2022-08-09T13:22:30Z"
+---
+
+Summary: Plugin system for spin.
+
+Owner: karthik.ganeshram@fermyon.com
+
+Created: August 9, 2022 
+
+## Background 
+
+As the functionality fo spin gets extended, there will be a point where every feature will not be required/used by every user. An example of this would be different triggers for spin along with possible subcommands to add more functionality to spin. Therefore it would make sense to be able to add features as plugins allowing for addition of only the required features without modifying the spin binary.
+
+## Proposal
+
+Create a new subcommand for spin called `spin plugin` which will have two further subcommands namely
+
+- `install`
+- `uninstall`
+
```

Comment:
I suspect that "upgrade" is something we'll want pretty early. Being able to upgrade `spin-play` will be important.

Reply:
Never mind, comment thread below.